Lib/doctest.py

```diff
@@ -235,25 +236,32 @@ def _normalize_module(module, depth=2):
     else:
         raise TypeError("Expected a module, string, or None")
 
-def _newline_convert(data):
-    # The IO module provides a handy decoder for universal newline conversion
-    return IncrementalNewlineDecoder(None, True).decode(data, True)
-
 def _load_testfile(filename, package, module_relative, encoding):
     if module_relative:
         package = _normalize_module(package, 3)
+        try:
+            loader = package.__loader__
+        except AttributeError:
+            pass
+        else:
+            if loader is not None:
+                return (
+                    importlib.resources.read_text(package, filename, encoding=encoding),
+                    filename,
+                )
+
+        try:
+            package.__spec__.loader
+        except AttributeError:
+            pass
+        else:
+            return (
+                importlib.resources.read_text(package, filename, encoding=encoding),
+                filename,
+            )
```

Comment:
`mod.__loader__` [is deprecated](https://docs.python.org/3/reference/datamodel.html#module.__loader__) in favour of `mod.spec.__loader__`, should we switch the order here? It may also be worth factoring the logic to a helper function.